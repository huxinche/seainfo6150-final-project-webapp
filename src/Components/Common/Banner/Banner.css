.banner{
    /* max-height: 400px; */
    overflow: hidden;
    color: white;
    position: relative;
}
.banner-img{
    width: 100%;
}

.banner-content{
    padding: 30px;
    position: absolute;
    text-align: center;
    top: 0;
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    height: 100%;
    z-index: 2;
    align-content: center;
    align-items: center;
}

.divider{
    margin: 20px auto;
    border-bottom: 10px solid rgb(151, 4, 4);
  
}

/* Extra small devices (phones, 600px and down) */
@media only screen and (max-width: 600px) {
    .divider{
        width: 100px;
    }
    .banner,.banner-content{
        max-height: 400px;
     }
}

/* Small devices (portrait tablets and large phones, 600px and up) */
@media only screen and (min-width: 600px) {
    .divider{
        width: 150px;
    }
    .banner,.banner-content {
        max-height: 450px;
     }
}

/* Medium devices (landscape tablets, 768px and up) */
@media screen and (min-width: 768px) {
    .divider{
        width: 200px;
    }
    .banner, .banner-content {
       max-height: 500px;
    }

}


