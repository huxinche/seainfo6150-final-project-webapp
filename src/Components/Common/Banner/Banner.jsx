import React from 'react';
import './Banner.css';
import PropTypes from "prop-types";

const Banner = (props) => {
    const { image, title, subtitle, divider } = props;
    return <div className="banner">
        <div>
            <picture className="banner-img">
                <img
                    src={image}
                    className="banner-img" alt={title} />
            </picture>
        </div>
        <div className="banner-content">
            <div >
                <h1 className="text-title">{title}</h1>
                {
                    divider && <div className="divider"></div>
                }
                <p className="text-title">{subtitle}</p>
            </div>
        </div>
    </div>
}

Banner.propTypes = {
    image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    divider: PropTypes.bool
}

export default Banner;