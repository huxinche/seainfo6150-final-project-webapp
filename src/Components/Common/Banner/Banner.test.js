import React from "react";
import Banner from "./Banner";

it("renders the Banner component correctly", () => {
    const { container } = render(<Banner image={'https://www.northeastern.edu/graduate/wp-content/uploads/2016/05/homepage_science.jpg'} title={'Expolre, Expand, Connect'}/>);
    expect(container).toMatchSnapshot();
})