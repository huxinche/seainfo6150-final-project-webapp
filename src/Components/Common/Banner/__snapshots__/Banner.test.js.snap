// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Banner component correctly 1`] = `
<div>
  <div
    class="banner"
  >
    <div>
      <picture
        class="banner-img"
      >
        <img
          alt="Expolre, Expand, Connect"
          class="banner-img"
          src="https://www.northeastern.edu/graduate/wp-content/uploads/2016/05/homepage_science.jpg"
        />
      </picture>
    </div>
    <div
      class="banner-content"
    >
      <div>
        <h1
          class="text-title"
        >
          Expolre, Expand, Connect
        </h1>
        <p
          class="text-title"
        />
      </div>
    </div>
  </div>
</div>
`;
